export.js

```diff
@@ -60,8 +61,9 @@ class SHR {
     }
 
     for (const de of cimcore.dataElements) {
-      this.elements.add(de);
-      const element = this.elements.get(de.fqn);
+      const deClone = Object.assign({}, de);
```

Comment:
`Object.assign` is not a deep clone -- it copies over the property values, but doesn't deeply copy them.  So if you add/remove properties from `deClone`, the original won't be modified, but if you mutate an existing property in `deClone`, it will be mutated in the original.  (In other words, `deClone.foo.bar = true` would modify the `foo.bar` property in both the clone and the original.

Are the `cimcore.dataElements` just JSON objects?  If so, you can do a deep copy via this:
```js
const deClone = JSON.parse(JSON.stringify(de));
```

If they are shr-models objects, then can't you just use `de.clone()`?

Reply:
Ah good catch. They aren't shr-models objects, rather CIMCORE json. (The Javadoc works through the CIMCORE.)

I'm actually surprised this worked in the first place as I did some tests to see if the mutations on properties still occurred.